l1-contracts-foundry/script/DeployL1.s.sol

```diff
@@ -0,0 +1,703 @@
+    }
+
+    struct StateTransitionDeployedAddresses {
+        address stateTransitionProxy;
+        address stateTransitionImplementation;
+        address verifier;
+        address adminFacet;
+        address mailboxFacet;
+        address executorFacet;
+        address gettersFacet;
+        address diamondInit;
+        address genesisUpgrade;
+        address defaultUpgrade;
+        address diamondProxy;
+    }
+
+    struct BridgesDeployedAddresses {
+        address erc20BridgeImplementation;
+        address erc20BridgeProxy;
+        address sharedBridgeImplementation;
+        address sharedBridgeProxy;
+    }
+
+    struct Config {
+        uint256 chainId;
+        uint256 eraChainId;
+        address deployerAddress;
+        uint256 gasPrice;
+        ContractsConfig contracts;
+        TokensConfig tokens;
+    }
+
+    struct ContractsConfig {
+        bytes32 create2FactorySalt;
+        address create2FactoryAddr;
+        uint256 validatorTimelockExecutionDelay;
+        bytes32 genesisRoot;
+        uint256 genesisRollupLeafIndex;
+        bytes32 genesisBatchCommitment;
+        uint256 latestProtocolVersion;
+        bytes32 recursionNodeLevelVkHash;
+        bytes32 recursionLeafLevelVkHash;
+        bytes32 recursionCircuitsSetVksHash;
+        uint256 priorityTxMaxGasLimit;
+        uint256 sharedBridgeUpgradeStorageSwitch;
+        uint256 diamondInitBatchOverheadL1Gas;
+        uint256 diamondInitMaxPubdataPerBatch;
+        uint256 diamondInitMaxL2GasPerBatch;
+        uint256 diamondInitPriorityTxMaxPubdata;
+        uint256 diamondInitMinimalL2GasPrice;
+        address governorAddress;
+        address governanceSecurityCouncilAddress;
+        uint256 governanceMinDelay;
+    }
+
+    struct TokensConfig {
+        address tokenWethAddress;
+    }
+
+    Config config;
+    DeployedAddresses addresses;
+
+    function run() public {
+        console.log("Deploying L1 contracts");
+
+        initializeConfig();
+
+        instantiateCreate2Factory();
+        deployIfNeededMulticall3();
+
+        deployVerifier();
+
+        deployDefaultUpgrade();
+        deployGenesisUpgrade();
+        deployValidatorTimelock();
+
+        deployGovernance();
+        deployTransparentProxyAdmin();
+        deployBridgehubContract();
+        deployBlobVersionedHashRetriever();
+        deployStateTransitionManagerContract();
+        setStateTransitionManagerInValidatorTimelock();
+
+        deployDiamondProxy();
+
+        deployErc20BridgeProxy();
+        deploySharedBridgeContracts();
+        deployErc20BridgeImplementation();
+        upgradeL1Erc20Bridge();
+
+        saveOutput();
+    }
+
+    function initializeConfig() internal {
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(root, "/script-config/config-deploy-l1.toml");
+        string memory toml = vm.readFile(path);
+
+        config.chainId = block.chainid;
+        config.deployerAddress = msg.sender;
```

Comment:
what will be the value of `msg.sender`? I couldn't find in the documentation

Reply:
it's a deployer.